webui/react/src/pages/ExperimentList.tsx

```diff
@@ -79,6 +132,69 @@ const ExperimentList: React.FC = () => {
     handleFilterChange({ ...filters, username });
   }, [ filters, handleFilterChange ]);
 
+  const getBatchActions = useCallback((action: Action): Promise<void[] | Command> => {
+    if (action === Action.OpenTensorBoard) {
+      return launchTensorboard({
+        ids: selectedExperiments.map(experiment => experiment.id),
+        type: TBSourceType.Experiment,
+      });
+    }
+    return Promise.all(selectedExperiments
+      .map(experiment => {
+        if (action === Action.Activate) {
+          return setExperimentState({ experimentId: experiment.id, state: RunState.Active });
+        } else if (action === Action.Archive) {
+          return archiveExperiment(experiment.id, true);
+        } else if (action === Action.Cancel) {
+          return setExperimentState({ experimentId: experiment.id, state: RunState.Canceled });
+        } else if (action === Action.Kill) {
+          return killExperiment({ experimentId: experiment.id });
+        } else if (action === Action.Pause) {
+          return setExperimentState({ experimentId: experiment.id, state: RunState.Paused });
+        }
+        return Promise.resolve();
+      }));
+  }, [ selectedExperiments ]);
+
+  const handleBatchAction = useCallback(async (action: Action) => {
+    try {
+      const result = await getBatchActions(action);
+      if (action === Action.OpenTensorBoard) {
+        const url = waitPageUrl(result as Command);
+        if (url) openBlank(setupUrlForDev(url));
+      }
+      await fetchExperiments();
+    } catch (e) {
+      const publicSubject = action === Action.OpenTensorBoard ?
+        'Unable to Open TensorBoard for Selected Experiments' :
+        `Unable to ${action} Selected Experiments`;
+      handleError({
+        error: e,
+        level: ErrorLevel.Error,
+        message: e.message,
+        publicMessage: 'Please try again later.',
+        publicSubject,
+        silent: false,
+        type: ErrorType.Server,
+      });
+    }
+  }, [ getBatchActions, fetchExperiments ]);
+
+  const handleConfirmation = useCallback((action: Action) => {
+    Modal.confirm({
```

Comment:
blocker: did you checkout antd https://ant.design/components/popconfirm/ ? let's use that unless you want to use the fact that modal blocks the whole screen.

Reply:
discussed async to keep Modal so we can block the rest of the UI while batch operations are occuring.